main.cpp

```diff
@@ -1,37 +1,74 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <sstream>
+#include <cassert>
 
-bool test_mode = true;
+namespace colors
+{
+    char red[] = "\033[31m";
+    char blue[] = "\033[34m";
+    char none[] = "\033[0m";
+}
+
+bool test_mode = false;
 
-int main()
+//requires minimum 3 arguments
+//maximum 4
+int main(int argc, char *argv[])
 {
-    std::string file_in_path = "./in.f", file_out_path = "./out.f";
-    /*
-    std::cout << "Name of in file:";
-    std::cin >> file_in_path;
-    std::cin.clear();
-    std::cin.ignore(std::numeric_limits<std::streamsize>::max(),'\n');
-    */
+    if(argc > 4 || argc < 3)
+    {
+        std::cout << colors::red << "Got " << argc << " arguments, but 3 or 4 needed" <<
+        std::endl << "Terminating" << colors::none << std::endl;
+        return 1;
+    }
+    std::string file_in_path = argv[1], file_out_path = argv[2];
+
+    if(file_in_path == file_out_path)
+    {
+        std::cout << colors::red << "AAAA Fuck!" << colors::none << std::endl;
+        file_out_path = "out.f";
+        std::cout << "FILE OUT: " << file_out_path << std::endl;
+    }
 
     std::ifstream fin(file_in_path,std::ios::binary);
     std::ofstream fout(file_out_path,std::ios::binary);
 
     fin.seekg(0,std::ios::end);
-    int size = fin.tellg();
+    int file_size = fin.tellg();
     fin.seekg(0,std::ios::beg);
 
-    if(test_mode)
+    std::cout << "size of in file: " << file_size << " byte" << std::endl;
+
+    int unit_size;
+    try
+    {
+        std::stringstream convert(
```

Comment:
You commented that convert is initialized by argv[1], but argv[3] used in the code